kobo/apps/subsequences/api_view.py

```diff
@@ -48,8 +49,17 @@ def _check_asr_mt_access_if_applicable(user, posted_data):
                     raise PermissionDenied('ASR/MT features are not available')
 
 
+class AdvancedSubmissionPermission(SubmissionPermission):
+    """
+    Regular `SubmissionPermission` maps POST to `add_submissions`, but
+    `change_submissions` should be required here
+    """
+    perms_map = deepcopy(SubmissionPermission.perms_map)
+    perms_map['POST'] = ['%(app_label)s.change_%(model_name)s']
+
```

Comment:
I know if only few lines but I would have preferred to keep the permission class in its own file (to be consistent with the rest of the app).

Reply:
Yeah, I thought about it. I didn't want to make a tiny file in the `subsequences` app, and I wasn't sure about putting it in the same file as `SubmissionPermission`, where the name "advanced" might be confusing to read without the context of `subsequences`. For best readability, I would just leave it here next to the only thing that uses it, but if more extensive `subsequences` permission work is needed in the future, or the permissions class gets used in other places, then I'd be in favor of moving to a separate file.